File: booking-backend/src/main/java/com/booking/service/BookingService.java
package com.booking.service;

import com.booking.dto.request.BookingRequest;
import com.booking.dto.response.BookingResponse;

import java.util.List;

public interface BookingService {

    BookingResponse createBooking(Long clientId, BookingRequest request);

    List<BookingResponse> getMyBookings(Long clientId);

    void cancelBooking(Long bookingId, Long clientId);

    List<BookingResponse> getProviderBookings(Long providerUserId);

    BookingResponse completeBooking(Long bookingId, Long providerUserId);

    void cancelBookingByProvider(Long bookingId, Long providerUserId);
}

File: booking-frontend/src/app/core/services/booking.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { BookingRequest, BookingResponse } from '../models/booking.model';

@Injectable({ providedIn: 'root' })
export class BookingService {
  private readonly baseUrl = `${environment.apiUrl}/bookings`;

  constructor(private http: HttpClient) {}

  createBooking(request: BookingRequest): Observable<BookingResponse> {
    return this.http.post<BookingResponse>(this.baseUrl, request);
  }

  getMyBookings(): Observable<BookingResponse[]> {
    return this.http.get<BookingResponse[]>(`${this.baseUrl}/my`);
  }

  cancelBooking(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
